src/ast.c

```diff
@@ -326,9 +326,14 @@ static jl_value_t *scm_to_julia_(value_t e, int eo)
 
             e = cdr_(e);
             if (!eo) {
-                if (sym == line_sym && n==1) {
-                    return jl_new_struct(jl_linenumbernode_type,
-                                         scm_to_julia_(car_(e),0));
+                if (sym == line_sym && n==2) {
+                    jl_value_t *filename = scm_to_julia_(car_(cdr_(e)),0);
+                    jl_value_t *linenum  = scm_to_julia_(car_(e),0);
+                    JL_GC_PUSH2(&filename, &linenum);
+                    jl_value_t *temp = jl_new_struct(jl_linenumbernode_type,
+                                                     filename, linenum);
+                    JL_GC_POP();
+                    return temp;
                 }
                 if (sym == label_sym) {
                     return jl_new_struct(jl_labelnode_type,
```

Comment:
:person_frowning: yes all of these need are missing a gc root for their argument